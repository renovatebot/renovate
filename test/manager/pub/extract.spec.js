const fs = require('fs');
const { extractPackageFile } = require('../../../lib/manager/pub');

const brokenYaml = fs.readFileSync(
  'test/manager/pub/_fixtures/update.yaml',
  'utf8'
);

const packageFile = fs.readFileSync(
  'test/manager/pub/_fixtures/extract.yaml',
  'utf8'
);

describe('manager/pub', () => {
  describe('extractPackageFile', () => {
    it('should return null if package does not contain any deps', () => {
      const res = extractPackageFile('foo: bar', 'pubspec.yaml');
      expect(res).toEqual(null);
    });
    it('should return null if package is invalid', () => {
      const res = extractPackageFile(brokenYaml, 'pubspec.yaml');
      expect(res).toEqual(null);
    });
    it('should return valid dependencies', () => {
      const res = extractPackageFile(packageFile, 'pubspec.yaml');
      expect(res).toMatchSnapshot();
    });
  });
});
