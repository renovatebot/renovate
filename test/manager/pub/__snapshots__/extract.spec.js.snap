// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manager/pub extractPackageFile should return valid dependencies 1`] = `
Object {
  "datasource": "dart",
  "deps": Array [
    Object {
      "currentValue": "1",
      "depName": "foo",
      "depType": "dependencies",
    },
    Object {
      "currentValue": "1",
      "depName": "bar",
      "depType": "dependencies",
    },
    Object {
      "currentValue": null,
      "depName": "baz",
      "depType": "dependencies",
      "skipReason": "not-a-version",
    },
    Object {
      "currentValue": null,
      "depName": "qux",
      "depType": "dependencies",
      "skipReason": "not-a-version",
    },
    Object {
      "currentValue": "^0.1",
      "depName": "test",
      "depType": "dev_dependencies",
    },
    Object {
      "currentValue": "0.1",
      "depName": "build",
      "depType": "dev_dependencies",
    },
  ],
  "manager": "pub",
  "packageFile": "pubspec.yaml",
}
`;
